app/code/community/Ebizmarts/MailChimp/Model/Observer.php

```diff
@@ -210,16 +210,19 @@ public function subscriberSaveAfter(Varien_Event_Observer $observer)
         $storeViewId = $this->getStoreViewIdBySubscriber($subscriber);
         $helper = $this->makeHelper();
         $isEnabled = $helper->isSubscriptionEnabled($storeViewId);
+        $subscriberSource = $subscriber->getSubscriberSource();
 
-        if ($isEnabled && $subscriber->getSubscriberSource() != Ebizmarts_MailChimp_Model_Subscriber::SUBSCRIBE_SOURCE) {
+        if ($isEnabled && $subscriberSource !== Ebizmarts_MailChimp_Model_Subscriber::SUBSCRIBE_SOURCE) {
```

Comment:
Extract second condition to function called isMailchimpSave to know what that constant means.